File: app/src/main/java/com/pkfcooperparry/template/ui/form/FormActivity.java
package com.pkfcooperparry.template.ui.form;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.GridLayoutManager;

import com.pkfcooperparry.template.BR;
import com.pkfcooperparry.template.R;
import com.pkfcooperparry.template.data.model.db.visit.Page;
import com.pkfcooperparry.template.databinding.ActivityFormBinding;
import com.pkfcooperparry.template.ui.base.BaseActivity;
import com.pkfcooperparry.template.ui.custom.CustomDialog;
import com.pkfcooperparry.template.ui.form.page.PageAdapter;
import com.pkfcooperparry.template.utils.Navigator;

import javax.inject.Inject;

public class FormActivity extends BaseActivity<ActivityFormBinding, FormViewModel> implements FormViewInterface {

    @Inject
    CustomDialog customDialog;

    @Inject
    Navigator navigator;

    @Inject
    FormViewModel formViewModel;

    ActivityFormBinding activityFormBinding;

    public static Intent getCallingIntent(@NonNull Context context, int formId) {
        Intent intent = new Intent(context, FormActivity.class);
        intent.putExtra("formId", formId);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        formViewModel.setViewInterface(this, this);
        activityFormBinding = getViewDataBinding();

        activityFormBinding.recyclerView.setLayoutManager(new GridLayoutManager(this, 5));
        PageAdapter pageAdapter = new PageAdapter(this, new PageAdapter.ItemClickListener() {
            @Override
            public void onItemClick(Page view) {
                navigator.navigateToPageView(FormActivity.this, (int) view.getPageId());
            }

        }, formViewModel.getFormWithId(getIntent().getIntExtra("formId", -1)).getPages());
        activityFormBinding.recyclerView.setAdapter(pageAdapter);
    }


    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_form;
    }

    @Override
    public FormViewModel getViewModel() {
        return formViewModel;
    }

    @Override
    public void handleError(String message) {
        customDialog.showDialog(this, CustomDialog.ONE_BUTTON_DIALOG_STYLE, getString(R.string.something_went_wrong), message);

    }

    @Override
    public void goBack() {
        finish();
    }

    @Override
    public void openRequests() {

    }

    @Override
    public void openActionPlan() {

    }

    @Override
    protected String getActionBarTitle() {
        return null;
    }

    @Override
    protected int getActionBarType() {
        return KEY_NO_ACTION_BAR;

    }

}

File: app/src/main/java/com/televantou/template/ui/form/FormActivity.kt
package com.televantou.template.ui.form

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager

import com.televantou.template.BR
import com.televantou.template.R
import com.televantou.template.databinding.ActivityFormBinding
import com.televantou.template.ui.base.BaseActivity
import com.televantou.template.ui.custom.CustomDialog
import com.televantou.template.utils.Navigator

import javax.inject.Inject

class FormActivity : BaseActivity<ActivityFormBinding, FormViewModel>(), FormViewInterface {

   var customDialog: CustomDialog? = null
        @Inject   set
     var navigator: Navigator? = null
        @Inject   set

    override var viewModel: FormViewModel? = null
        @Inject   set

    internal var activityFormBinding: ActivityFormBinding? = null


    override val bindingVariable: Int
        get() = BR.viewModel

    override val layoutId: Int
        get() = R.layout.activity_form

    protected override val actionBarTitle: String?
        get() = null

    protected override val actionBarType: Int
        get() = BaseActivity.KEY_NO_ACTION_BAR

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel!!.setViewInterface(this, this)
        activityFormBinding = viewDataBinding

        activityFormBinding!!.recyclerView.layoutManager = GridLayoutManager(this, 5)
//        val pageAdapter = viewModel!!.getFormWithId(intent.getIntExtra("formId", -1))!!.pages?.let {
//            PageAdapter(this, PageAdapter.ItemClickListener {
//            override fun onItemClick(page: Page) {
//                navigator!!.navigateToPageView(this@FormActivity, page.pageId.toInt())
//            }
//        }, it)
//        }
//        activityFormBinding!!.recyclerView.adapter = pageAdapter
    }

    override fun handleError(message: String) {
        customDialog!!.showDialog(this, CustomDialog.ONE_BUTTON_DIALOG_STYLE, getString(R.string.something_went_wrong), message)

    }

    override fun goBack() {
        finish()
    }

    override fun openRequests() {

    }

    override fun openActionPlan() {

    }

    companion object {

        fun getCallingIntent(context: Context, formId: Int): Intent {
            val intent = Intent(context, FormActivity::class.java)
            intent.putExtra("formId", formId)
            return intent
        }
    }

}
